config/initializers/folio_sync.rb

```diff
@@ -1,4 +1,37 @@
 # frozen_string_literal: true
 
-# Ensure that MARC download directory exists
-FileUtils.mkdir_p(Rails.configuration.folio_sync['marc_download_directory'])
+# Ensure that MARC download directories exist
+begin
+  logger = Logger.new($stdout)
+  folio_config = Rails.configuration.folio_sync
+
+  if folio_config.blank?
+    logger.error('FOLIO sync configuration is missing or empty')
+    throw 'Please make sure the folio_sync.yml file is present and properly configured.'
+  end
+
+  instances = folio_config['instances'] || {}
+
+  if instances.empty?
+    logger.error('No instances configured')
+    throw 'Please make sure the folio_sync.yml file contains instances'
+  end
+
+  base_dir = folio_config['marc_download_base_directory']
+
+  # If the base directory is not present, set a default base direcotry
+  if base_dir.blank?
+    current_env = Rails.env
+    base_dir = Rails.root.join("tmp/#{current_env}/downloaded_files")
```

Comment:
Imo we should have a default directory for downloaded files (as long as instances are specified in the `folio_sync.yml`).

Reply:
Creating a default download directory definitely works, but just offering this suggestion in case it simplifies things:

`raise 'Missing required folio_sync.yml configuration option: marc_download_base_directory' if base_dir.blank?`

Putting this in an initializer will prevent the rails app from starting up, and will prevent the rake task from running, but I think that's completely fine if an important config option is missing.  The error message would make it clear what's wrong.